Fixes Persistence not spawning on icebox (also adds a roof so it stops having holes) (ready)

shellspeed1 opened the pull request:
## About The Pull Request
(new version of original PR because github was being really annoying, see original here https://github.com/Bubberstation/Bubberstation/pull/3595)

First of all, credit to https://github.com/IgiariValkyr (Otome on discord or Aster Briar in game) for getting the initial stuff setup for the map and giving me the basis for this PR. It saved me a ton of time and effort and turned this from a day long project to just a couple of hours.

This PR turns persistence back into a static spawn ruin and also adds a roof area above it to prevent having holes in the roof.

The roof itself does not feature notable items. As for its significant areas it does have the main roof access hatch, six ship clamps for moving the ship (fluff), additional cargo storage with a hopefully soon to be working elevator, small grill, logos, and two watch posts.

On a more contentious note, I have added a shield generator setup as seen on previous iterations of the interdyne map in lieu of significant turret placements (There are still turrets). This is something I am testing as an alternative to turrets as there has been complaints about the turrets being a bit much. There are still two turrets in the main access hatch and the shield does need to be setup to function which can be done by multitooling all of the parts together (takes about 3 minutes).

Additionally, cleaned up some additional items like a bottle of whiskey someone added with my ckey, making sure botany had space for a bee hive,

To Do:
Add: moon station version

## Why It's Good For The Game

I adding a shipping container ranch and no one can stop me. Even worse, I added an elevator.

![image](https://github.com/user-attachments/assets/f0d4c199-4ed7-4651-8ab5-e50ee8ba4f44)

On a more serious note, I have had several complaints about persistence not spawning correctly due to a lack of safe areas to do it, so I am properly fixing it this time.
## Proof Of Testing

https://youtu.be/0YoZErQ7cOA
<details>
<summary>Screenshots/Videos</summary>

</details>

## Changelog
:cl:
add: Persistence now has an accessible roof on icebox
fix: Swapped scientist modsuit to infiltrator so they could use provided mod module.
fix: Persistence is now a static spawn again
map: The persistence janitor has a bedroom again
/:cl:

NargaInDisguise commented:
I really do not agree with the shield generator over the turrets. 
Also uh, the persistence makeover got merged, so uh, there might be a lot of merge issues.

shellspeed1 commented:
> I really do not agree with the shield generator over the turrets.
> 
> Ideally there should be literally no core differences between persistence on all maps. Having a roof area on icebox was supposed to be the exemption solely because its required to make it a static spawn.
> 
> Also uh, the persistence makeover got merged, so uh, there might be a lot of merge issues.

I literally took the new maps and redid it because of all the merge conflicts

shellspeed1 commented:
> Hey, a reorganization of the automapper file/template folders just got merged so I had to manually merge master on your PR here, hope you don't mind, just moved and renamed the template out of the RandomRuins folder and updated the config

No problem, life like usual is chaos for me here.

R0bro commented:
While I love the idea of percy having a roof I feel as it stands it actually invites more nefarious deeds. For starters, there are no cameras or turrets on the upper floor making it fairly easy to just waltz on top of it without anyone even noticing. The shield, while also cool and trying to make up for the lack of turrets, doesn't span the entire area of the roof. 

Additional things I noticed/have an opinion on: 
- Since there are a ton of tiles in a large flat area it's very easy to get multiple smuggler's satchels, depriving poor Nanotrasen of them. Preferably the use of reinforced hull plating tiles would fix this? Not susre.
- The roof has far too many lights. If it were me I'd marker beacons to give it more of a mood and keep the extra flood lights if crew want the extra visibility.
- There is an excessive amount of traversable space and not enough detail in my opinion. I'd love if there were a few more structures dotted about the roof as well as more snow, way more snow.
- The elevator doesn't work by default. Is this intentional?

shellspeed1 commented:
> While I love the idea of percy having a roof I feel as it stands it actually invites more nefarious deeds. For starters, there are no cameras or turrets on the upper floor making it fairly easy to just waltz on top of it without anyone even noticing. The shield, while also cool and trying to make up for the lack of turrets, doesn't span the entire area of the roof.
> 
> Additional things I noticed/have an opinion on:
> 
> * Since there are a ton of tiles in a large flat area it's very easy to get multiple smuggler's satchels, depriving poor Nanotrasen of them. Preferably the use of reinforced hull plating tiles would fix this? Not sure.
> * The roof has far too many lights. If it were me I'd marker beacons to give it more of a mood and keep the extra flood lights if crew want the extra visibility.
> * There is an excessive amount of traversable space and not enough detail in my opinion. I'd love if there were a few more structures dotted about the roof as well as more snow, way more snow.
> * The elevator doesn't work by default. Is this intentional?

I'll take a look at the smuggler satchel issue and see if I can figure out something there, the reinforced hull plating tiles are exactly pretty but I have some new tricks to get around that if need be. As for lights, that is a fair item but its also meant to be a large work site area. It _SHOULD_ be well lit ideally. The lack of turrets is because I've had complaints in the past of having too many turrets and the turrets as is being too strong. The shield is upgradable to provide full roof coverage and is fairly tough if upgraded. There are also still turrets in the entry hatches. Lastly, the elevator is only on Icebox atm, no other maps will have it for another day or two.

NargaInDisguise commented:
> > While I love the idea of percy having a roof I feel as it stands it actually invites more nefarious deeds. For starters, there are no cameras or turrets on the upper floor making it fairly easy to just waltz on top of it without anyone even noticing. The shield, while also cool and trying to make up for the lack of turrets, doesn't span the entire area of the roof.
> > Additional things I noticed/have an opinion on:
> > 
> > * Since there are a ton of tiles in a large flat area it's very easy to get multiple smuggler's satchels, depriving poor Nanotrasen of them. Preferably the use of reinforced hull plating tiles would fix this? Not sure.
> > * The roof has far too many lights. If it were me I'd marker beacons to give it more of a mood and keep the extra flood lights if crew want the extra visibility.
> > * There is an excessive amount of traversable space and not enough detail in my opinion. I'd love if there were a few more structures dotted about the roof as well as more snow, way more snow.
> > * The elevator doesn't work by default. Is this intentional?
> 
> I'll take a look at the smuggler satchel issue and see if I can figure out something there, the reinforced hull plating tiles are exactly pretty but I have some new tricks to get around that if need be. As for lights, that is a fair item but its also meant to be a large work site area. It _SHOULD_ be well lit ideally. The lack of turrets is because I've had complaints in the past of having too many turrets and the turrets as is being too strong. The shield is upgradable to provide full roof coverage and is fairly tough if upgraded. There are also still turrets in the entry hatches. Lastly, the elevator is only on Icebox atm, no other maps will have it for another day or two.

I still don't understand the arguement about "the turrets are too strong"

THEY'RE SUPPOSED to be strong and without any blindspots. They're supposed to keep players AWAY from it. They're hidden behind an entire rock wall that has an obvious tell with the plasma/lava lake.
If they're doing their job I don't get the problem...
It's either that or we just start fucking doming people that stray a little too close ourselves, which isn't an alternative that people would like either.

shellspeed1 commented:
![image](https://github.com/user-attachments/assets/34a4f58b-7526-4985-9d07-73ac86216f8b)

Down to 1,000 when the maps loads, still working through it although all the outside tiles should have correct atmos.

shellspeed1 commented:
![image](https://github.com/user-attachments/assets/3221b663-4aef-46a9-a382-3fb3c2acbeb9)

More progress
